multimutex/multimutex_hash.go

```diff
@@ -0,0 +1,90 @@
+package multimutex
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+// HashMutex is a struct that keeps track of a set of mutexes with
```

Comment:
Entire file appears to be folded to less than 80 characters for each column.

Reply:
An alternative to this approach could be to:  
  1. Modify the base `multimutex` to be based on a `string` key
  2. Update the current integer based version to map to a string
  3. Use the base version in place of this 

However, I think we have a few callers of this package in `neutrino`, so then we'd need to update them all as well.

Reply:
Folding fixed.

Converted to string to prevent more of these types emerging in the future. Leaving int based one in place. It is probably more efficient and also to not let this pr sprawl into unrelated dependents of multimutex.